src/VkBootstrap.h

```diff
@@ -757,7 +811,31 @@ class DeviceBuilder {
 
     // For Advanced Users: specify the exact list of VkDeviceQueueCreateInfo's needed for the application.
     // If a custom queue setup is provided, getting the queues and queue indexes is up to the application.
-    DeviceBuilder& custom_queue_setup(std::vector<CustomQueueDescription> queue_descriptions);
+    DeviceBuilder& custom_queue_setup(size_t count, CustomQueueDescription const* queue_descriptions) {
+        info.queue_descriptions.assign(queue_descriptions, queue_descriptions + count);
+        return *this;
+    }
+    // For Advanced Users: specify the exact list of VkDeviceQueueCreateInfo's needed for the application.
+    // If a custom queue setup is provided, getting the queues and queue indexes is up to the application.
+    DeviceBuilder& custom_queue_setup(std::vector<CustomQueueDescription> const& queue_descriptions) {
+        info.queue_descriptions = queue_descriptions;
+        return *this;
+    }
+    // For Advanced Users: specify the exact list of VkDeviceQueueCreateInfo's needed for the application.
+    // If a custom queue setup is provided, getting the queues and queue indexes is up to the application.
+    DeviceBuilder& custom_queue_setup(std::vector<CustomQueueDescription>&& queue_descriptions) {
+        info.queue_descriptions = std::move(queue_descriptions);
+        return *this;
+    }
+
+#if VKB_SPAN_OVERLOADS
+    // For Advanced Users: specify the exact list of VkDeviceQueueCreateInfo's needed for the application.
+    // If a custom queue setup is provided, getting the queues and queue indexes is up to the application.
+    DeviceBuilder& custom_queue_setup(std::span<const CustomQueueDescription> queue_descriptions) {
+        info.queue_descriptions.assign(queue_descriptions.begin(), queue_descriptions.end());
+        return *this;
```

Comment:
Same as the InstanceBuilder, would much prefer implementations to be in the .cpp